components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/ArgumentProcessor.kt

```diff
@@ -61,23 +63,47 @@ private fun createCommandLineArgs(intent: Intent): CliArgs? {
         // There is a quoting within quoting problem meaning apostrophes don't get sent
         // through the multiple shells properly. This steps around this issue completely.
         ?.replace("&apos;", "'")
-        ?.let { string ->
-            // We do some rudimentary taint checking of the string:
-            // we make sure it looks like a JSON object, with a `data` key
-            // and an array value.
-            try {
-                val jsonObject = JSONObject(string)
-                jsonObject.optJSONArray(DATA_KEY) ?: return@let null
-                string
-            } catch (e: JSONException) {
-                null
-            }
-        }
 
     val resetDatabase = intent.getBooleanExtra(RESET_DB_KEY, false)
     val logState = intent.getBooleanExtra(LOG_STATE_KEY, false)
 
-    return CliArgs(resetDatabase, experiments, logState)
+    return check(CliArgs(resetDatabase, experiments, logState))
+}
+
+@Suppress("ReturnCount")
+private fun check(args: CliArgs): CliArgs? {
+    // We do some rudimentary taint checking of the string:
+    // we make sure it looks like a JSON object, with a `data` key
+    // and an array value.
+    val string = args.experiments
+    if (string != null) {
+        try {
+            val jsonObject = JSONObject(string)
+            jsonObject.optJSONArray(DATA_KEY) ?: return null
+        } catch (e: JSONException) {
+            return null
+        }
+    }
+    return args
+}
+
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+@Suppress("ReturnCount")
+fun createCommandLineArgs(uri: Uri): CliArgs? {
+    if (setOf("http", "https").contains(uri.scheme ?: "")) {
+        return null
+    }
+    val isForUs = uri.getBooleanQueryParameter("--$NIMBUS_FLAG", false)
```

Comment:
Nit: This is `isForUs` here and `meantForUs` in the Swift code, I think it would be ideal to have them use the same name across both